src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,203 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    private static class Node<T> {
+        T element;
+        Node<T> next;
+        Node<T> previous;
+
+        public Node(Node<T> previous, T element, Node<T> next) {
+            this.previous = previous;
+            this.element = element;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            size = 0;
+            first = newNode;
+            last = first;
+        } else {
+            last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == 0) {
+            addFirst(value);
+        } else {
+            if (index == size) {
+                addToEnd(value);
+            } else {
+                if (isIndexExist(index)) {
+                    Node<T> newNode = new Node<>(null,value,null);
+                    Node<T> oldNode = findForIndex(index);
+                    Node<T> oldPrev = findForIndex(index - 1);
```

Comment:
```suggestion
                    Node<T> oldPrev = oldNode.prev;
```
you don't need to call findForIndex() twice here